app/controllers/meetings_controller.rb

```diff
@@ -259,29 +201,60 @@ def destroy
   private
 
   # Use callbacks to share common setup or constraints between actions.
-  # rubocop:disable RescueStandardError
   def set_meeting
     @meeting = Meeting.friendly.find(params[:id])
-  rescue
+  rescue ActiveRecord::RecordNotFound
     redirect_to_path(groups_path)
   end
-  # rubocop:enable RescueStandardError
 
   # Checks if user is a meeting leader, if not redirect to group_path
-  def not_a_leader(group_id)
-    group_member = GroupMember.where(
-      group_id: group_id,
-      user_id: current_user.id,
-      leader: true
-    )
-    return if group_member.exists?
+  def leader?(group)
+    return if GroupMember.find_by(
+      user_id: current_user.id, group_id: group.id, leader: true
+    ).present?
 
-    redirect_to_path(group_path(group_id))
+    redirect_to_path(group_path(group.id))
```

Comment:
I know this was here when you started this refactor, but can this redirect be part of the code path in the destroy method? This reads initially that the `leader?` method should return a boolean, so maybe have it just run the query in the guard clause and then the redirect logic can go in the destroy method